tools/ipapy2/ipapy2_MS1_annotation.xml

```diff
@@ -0,0 +1,105 @@
+<tool id="ipapy2_MS1_annotation" name="4. IPA MS1 annotation" version="@TOOL_VERSION@+galaxy0" python_template_version="3.5" profile="21.05">
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">ipapy2</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        python3 '${ipapy2_MS1_annotation_cli}'
+    ]]></command>
+
+<configfiles>
+<configfile name="ipapy2_MS1_annotation_cli">
+@init_logger@
+
+df = pd.read_csv('${MS1_table}')
+allAdds = pd.read_csv('${all_adducts}' )
+#if $ppmunk == "None"
+#if $ppmthr == "None"
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppm, ratiounk=$ratiounk, ppmthr=2*$ppm, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#else
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppm, ratiounk=$ratiounk, ppmthr=$ppmthr, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#end if
+#else
+#if $ppmthr == "None"
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppmunk, ratiounk=$ratiounk, ppmthr=2*$ppm, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
+#else
+annotations = ipa.MS1annotation(df, allAdds, ppm=$ppm, me=$me, ratiosd=$ratiosd, ppmunk=$ppmunk, ratiounk=$ratiounk, ppmthr=$ppmthr, pRTNone=$pRTNone, pRTout=$pRTout, ncores=$ncores)
```

Comment:
There should be only a single call to the annotation function and the variables should be initialized before the call, depending on the provided parameters.